alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/model/AggregationTrigger.kt

```diff
@@ -64,8 +65,10 @@ data class AggregationTrigger(
             .field(ID_FIELD, id)
             .field(NAME_FIELD, name)
             .field(SEVERITY_FIELD, severity)
+            .startObject(CONDITION_FIELD)
```

Comment:
To keep this a little cleaner, we could override `toXContent()` for `BucketSelectorExtAggregationBuilder` since it does extend `ToXContent` and move the object representation there.

Ex.
```
override fun toXContent(builder: XContentBuilder, params: ToXContent.Params): XContentBuilder {
    builder.startObject()
    this.internalXContent(builder, params)
    return builder.endObject()
}
```

Then the `toXContent` in `AggregationTrigger` can be simplified to just be `.field(CONDITION_FIELD, bucketSelector, params)` since the `field()` method will check to see if it's a `ToXContent` object and call `toXContent()` implicitly for us.

Reply:
Got it, looking at `AbstractPipelineAggregationBuilder` I see what you mean. Looks like that implementation of `toXContent()` is creating a named object. I'm fine with the current implementation then.